jubatus/core/fv_converter/datum_to_fv_converter_test.cpp

```diff
@@ -442,6 +445,102 @@ TEST(datum_to_fv_converter, check_datum_key_in_binary) {
   ASSERT_THROW(conv.convert(datum, feature), converter_exception);
 }
 
+TEST(datum_to_fv_converter, combination_feature_num) {
+  datum datum;
+  datum.num_values_.push_back(std::make_pair("/val1", 1.1));
+  datum.num_values_.push_back(std::make_pair("/val2", 1.1));
+
+  datum_to_fv_converter conv;
+  init_weight_manager(conv);
+  typedef shared_ptr<combination_feature> combination_feature_t;
+  typedef shared_ptr<num_feature> num_feature_t;
+  shared_ptr<key_matcher> all_matcher(new match_all());
+
+  conv.register_num_rule(
+      "num",
+      all_matcher,
+      num_feature_t(new num_value_feature()));
+  conv.register_combination_rule(
+      "add",
+      all_matcher,
+      all_matcher,
+      combination_feature_t(new combination_add_feature()));
+  conv.register_combination_rule(
+      "mul",
+      all_matcher,
+      all_matcher,
+      combination_feature_t(new combination_mul_feature()));
+  std::vector<std::pair<std::string, float> > feature;
+  conv.convert(datum, feature);
+
+  std::vector<std::pair<std::string, float> > expected;
+  expected.push_back(std::make_pair("/val1@num", 1.1));
+  expected.push_back(std::make_pair("/val2@num", 1.1));
```

Comment:
How about using different number from `val1`. For confirmation that it is not using `val1` twice.
This test will be wrongly passed although we implementd 
```
void add_feature(const std::string& key,
  double value_left,
  double value_right,
  common::sfv_t& ret_fv) const {
  ret_fv.push_back(
   std::make_pair(key, static_cast<float>(value_left * value_left)));  // using left twice!!!!
}
```